aea/skills/base.py

```diff
@@ -900,15 +900,19 @@ def _filter_classes(
         - the class must be a subclass of "SkillComponent";
         - its __module__ attribute must not start with 'aea.' (we exclude classes provided by the framework)
         - its __module__ attribute starts with the expected dotted path of this skill.
+            In particular, it should not be imported from another skill.
 
         :param classes: a list of pairs (class name, class object)
         :return: a list of the same kind, but filtered with only skill component classes.
         """
         filtered_classes = filter(
             lambda name_and_class: issubclass(name_and_class[1], SkillComponent)
+            # the following condition filters out classes imported from 'aea'
             and not str.startswith(name_and_class[1].__module__, "aea.")
+            # the following condition filters out classes imported
+            # from other skills
             and not str.startswith(
-                name_and_class[1].__module__, self.skill_dotted_path
+                name_and_class[1].__module__, self.skill_dotted_path + "."
             ),
             classes,
         )
```

Comment:
these changes do not address the issue; are just for better documentation. Line 915 is good to avoid some corner cases in which the dotted path of a skill is a proper prefix of the current skill.